File: src/main/java/de/com/fdm/bot/commands/PartChannelCommand.java
package de.com.fdm.bot.commands;

import de.com.fdm.bot.Command;
import de.com.fdm.twitch.tmi.TmiMessage;
import de.com.fdm.twitch.tmi.TmiService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class PartChannelCommand implements Command {
    private final TmiService tmiService;

    public PartChannelCommand(@Autowired TmiService tmiService) {
        this.tmiService = tmiService;
    }
    @Override
    public String execute(TmiMessage tmiMessage) {
        if (tmiMessage.getArgs().size() == 0) {
            return "No channel provided";
        }

        String channel = tmiMessage.getArgs().get(0);
        tmiService.partChannel(channel);
        return "Parted channel #" + channel;
    }
}

File: src/main/java/de/com/fdm/bot/commands/PartChannelCommand.kt
package de.com.fdm.bot.commands

import de.com.fdm.bot.Command
import de.com.fdm.bot.api.twitch.tmi.TmiMessage
import de.com.fdm.bot.api.twitch.tmi.TmiReceiverService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class PartChannelCommand(@param:Autowired private val tmiReceiverService: TmiReceiverService) : Command {
    override fun execute(tmiMessage: TmiMessage): String {
        if (tmiMessage.getArgs().isEmpty()) {
            return "No channel provided"
        }

        val channel = tmiMessage.getArgs()[0]
        tmiReceiverService.partChannel(channel)
        return "Parted channel #$channel"
    }
}